TESTING

```diff
@@ -0,0 +1,404 @@
+// Include NeoPixel library
+#include <Adafruit_NeoPixel.h>
+
+// Define the PIN the strip is connected to, and the # of LEDs
+#define PIN      6
+#define N_LEDS 240
+
+// Define constant variables
+const int buttonPin = 2;
+const int buttonDelay = 150;
+
+// Define changing variables
+unsigned long previousMillis = 0;       // snapshot of time at the beginning of delay function
+unsigned long lastDebounceTime = 0;     // time of last button press
+int buttonState = 0;                    // whether button is HIGH or LOW
+int mode = 0;                           // the LED mode (currently 6 available)
+int x = 0;                              // used to change between colors in mode 1 and 2
+int counter = 0;                        // general counter
+uint32_t currentTime = 0;
+uint32_t milliseconds = 0;
+
+// Create a NeoPixel strip
+Adafruit_NeoPixel strip = Adafruit_NeoPixel(N_LEDS, PIN, NEO_GRB + NEO_KHZ800);
+
+// IMPORTANT: To reduce NeoPixel burnout risk, add 1000 uF capacitor across
+// pixel power leads, add 300 - 500 Ohm resistor on first pixel's data input
+// and minimize distance between Arduino and first pixel.  Avoid connecting
+// on a live circuit...if you must, connect GND first.
+
+
+void setup() {
+  // Initialize serial monitor communication
+  Serial.begin(9600);
+
+  // Initialize the pushbhtton pin as an input:
+  pinMode(buttonPin, INPUT);
+
+  strip.begin();
+  strip.show(); // Initialize all pixels to 'off'
+}
+
+
+void loop() {
+  uint32_t currentTime = millis();
+
+  // Christmas 2017 Cycle
+  LEDCrash();
+  for (int p = 0; p < 2; p++) {
+    if (p % 2 == 0) {
+    colorWipe(strip.Color(255, 0, random(0, 200)));
+    colorWipeBackwards(strip.Color(0, random(0, 200), 255));
+    colorWipe(strip.Color(random(0, 200), 255, 0));
+    } else {
+    colorWipeBackwards(strip.Color(255, 0, random(0, 200)));
+    colorWipe(strip.Color(0, random(0, 200), 255));
+    colorWipeBackwards(strip.Color(random(0, 200), 255, 0));
+    }
+    }
+  theaterChase(strip.Color(255, 0, 0), strip.Color(0, 255, 0));
```

Comment:
Add a mode check to the main loop function, it would look something like this (once you also create an enum for mode)
```

switch (mode) {
case PONG:
 pongBuild();
 break;
case RAINBOW:
 rainbowStack();
 break;
}
```
...